### ISSUE TITLE
Network: Auto zooming while stablizing on initial load.

### ISSUE THREAD

**Issue by dragonzone** [almende/vis#1266]
I've upgraded to v4.8 and notice that some of our users network charts tries to auto zoom in while the nodes and edges are stabilizing and won't stop until stabilization is done. I am not sure if is caused by the Kamada-Kawai new algorithm added (https://github.com/almende/vis/issues/1252 and https://github.com/almende/vis/issues/1255). I somehow cannot replicate on my machine but have seen on at least 2 of our users. So I want to bring it up to see if anyone else have this issue. 

Maybe to the same effect, I also set improvedLayout: false, but notice that on my machine that the nodes & edges are zoomed in very close on my network on initial load. I have to set .fit(). to get full network view.

**Comment by dragonzone** [comment]
Problem seem to be only in IE. Firefox seems to work fine.

**Comment by dragonzone** [comment]
This bug was hard to trace. But to replicate, make sure you use IE11. Go to the menu, select View | Zoom, and select 125%.

Go to the demo: http://visjs.org/examples/network/other/performance.html

Where it says "Number of nodes", change to 250 and press Go. Wait for a few seconds and you will see that it will keep on zooming smaller and smaller until it's gone.

**Comment by AlexDM0** [comment]
Hi Dragon,

I've had multiple reports on this and I'll try to get to the bottom of this asap. 

Sorry for the inconvenience. 

Regards

**Comment by closedLoop** [comment]
Experiencing similar issue on network graph in v4.8 (resolved using v4.7).  Every hover event and click event would shift the chart down and to the right a fixed amount.    Using Hierarchical 'LR' ordering and disabling any of the drag functionlity

Full (albiet not minimal options)
```javascript
var options = {
                nodes:{
                    borderWidth: 1,
                    borderWidthSelected: undefined,
                    brokenImage:undefined,
                    color: {
                        border: '#2B7CE9',
                        background: '#97C2FC',
                        highlight: {
                            border: '#2B7CE9',
                            background: '#D2E5FF'
                        },
                        hover: {
                            border: '#2B7CE9',
                            background: '#D2E5FF'
                        }
                    },
                    fixed: {
                        x:0,
                        y:0
                    },
                    font: {
                        color: '#343434',
                        size: 14, // px
                        face: 'arial',
                        background: 'none',
                        strokeWidth: 0, // px
                        strokeColor: '#ffffff'
                    },
                    group: undefined,
                    hidden: false,
                    image: undefined,
                    label: undefined,
                    labelHighlightBold: true,
                    level: undefined,
                    mass: 1,
                    physics: true,
                    scaling: {
                        min: 10,
                        max: 30,
                        label: {
                            enabled: false,
                            min: 14,
                            max: 30,
                            maxVisible: 30,
                            drawThreshold: 5
                        },
                        customScalingFunction: function (min,max,total,value) {
                            if (max === min) {
                                return 0.5;
                            }
                            else {
                                var scale = 1 / (max - min);
                                return Math.max(0,(value - min)*scale);
                            }
                        }
                    },
                    shadow:{
                        enabled: false,
                        size:10,
                        x:5,
                        y:5
                    },
                    shape: 'ellipse',
                    shapeProperties: {
                        borderDashes: false, // only for shapes with a border
                        borderRadius: 6 // only for box shape
                    },
                    size: 25
                },
                edges:{
                    arrows: {
                        to:     {enabled: true, scaleFactor:1},
                        middle: {enabled: false, scaleFactor:1},
                        from:   {enabled: false, scaleFactor:1}
                    },
                    color: {
                        color:'#848484',
                        highlight:'#848484',
                        hover: '#848484',
                        inherit: 'from',
                        opacity:1.0
                    },
                    dashes: false,
                    font: {
                        color: '#343434',
                        size: 14, // px
                        face: 'arial',
                        background: 'none',
                        strokeWidth: 2, // px
                        strokeColor: '#ffffff',
                        align:'horizontal'
                    },
                    hidden: false,
                    hoverWidth: 1.5,
                    label: undefined,
                    labelHighlightBold: true,
                    length: undefined,
                    physics: true,
                    scaling:{
                        min: 1,
                        max: 15,
                        label: {
                            enabled: true,
                            min: 14,
                            max: 30,
                            maxVisible: 30,
                            drawThreshold: 5
                        },
                        customScalingFunction: function (min,max,total,value) {
                            if (max === min) {
                                return 0.5;
                            }
                            else {
                                var scale = 1 / (max - min);
                                return Math.max(0,(value - min)*scale);
                            }
                        }
                    },
                    selectionWidth: 1,
                    selfReferenceSize:20,
                    shadow:{
                        enabled: false,
                        size:10,
                        x:5,
                        y:5
                    },
                    smooth: {
                        enabled: true,
                        type: "horizontal",
                        roundness: 0.0
                    },
                    width: 1
                },
                autoResize: true,
                width: '100%',
                height: '800px',
                layout: {hierarchical: {enabled: true, direction:'LR', sortMethod:'directed'}},
                interaction:{
                    dragNodes:false,
                    dragView: false,
                    hideEdgesOnDrag: false,
                    hideNodesOnDrag: false,
                    hover: true,
                    hoverConnectedEdges: true,
                    keyboard: {
                        enabled: false,
                        speed: {x: 10, y: 10, zoom: 0.02},
                        bindToWindow: true
                    },
                    multiselect: false,
                    navigationButtons: false,
                    selectable: true,
                    selectConnectedEdges: true,
                    tooltipDelay: 300,
                    zoomView: true
                },
                manipulation: {
                    enabled: true,
                    initiallyActive: true,
                    addNode: true,
                    addEdge: true,
                    editNode: undefined,
                    editEdge: true,
                    deleteNode: true,
                    deleteEdge: true,
                    controlNodeStyle:{
                        // all node options are valid.
                    }
                }
```

**Comment by dragonzone** [comment]
Hope this will help. If you turn on IE debugger and change to IE v10 or below, it'll work fine.

**Comment by AlexDM0** [comment]
Hi,

I'm looking into these issues, sorry for the inconvenience. Our last release was a little rushed due to scheduling. We'll take better care of this in future releases though bugs can always pop up. It's very hard to test a codebase like vis.

Thanks for reporting!

Cheers

**Comment by dragonzone** [comment]
No problem, this issue is a bit weird and hard to troubleshoot. Let me know how you resolve it. I'm very interested. Thx.

**Comment by AlexDM0** [comment]
It's due to both the blocking of the fit approximation since I expected the new algorithm would benefit more from a real range estimation instead of the approximation and the dynamic camera state consistency between changing canvas sizes. 

I'll have to rewrite parts of that but I'm currently very busy with helping out my colleagues with their Kickstarter campaign. I'm doing all the video work and 3D animations and there's a lot of deadlines there. 

Check it out! www.crownstone.rocks (shameless self promotion haha)

Cheers

**Comment by dragonzone** [comment]
Interesting, I might get one if I can afford it. :D

**Comment by AlexDM0** [comment]
We're aiming for most affordable on the market ;) sign up to be notified :)

Cheers

**Comment by AlexDM0** [comment]
Hi,

This should be fixed on develop now. Thanks for reporting! We'll release a bugfix release this week.

Cheers

**Comment by AlexDM0** [comment]
This has been fixed in 4.8.1

**Comment by aramse** [comment]
I'm seeing this same issue in 4.20.1. Anyone able to confirm the same?
